custom_components/tahoma/binary_sensor.py

```diff
@@ -1,64 +1,113 @@
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
 from .const import DOMAIN
+from .coordinator import TahomaDataUpdateCoordinator
 from .tahoma_entity import TahomaEntity
 
-CORE_ASSEMBLY_STATE = "core:AssemblyState"
-CORE_BUTTON_STATE = "core:ButtonState"
-CORE_CONTACT_STATE = "core:ContactState"
-CORE_GAS_DETECTION_STATE = "core:GasDetectionState"
-CORE_OCCUPANCY_STATE = "core:OccupancyState"
-CORE_OPENING_STATE = "core:OpeningState"
-CORE_OPEN_CLOSED_TILT_STATE = "core:OpenClosedTiltState"
-CORE_RAIN_STATE = "core:RainState"
-CORE_SMOKE_STATE = "core:SmokeState"
-CORE_THREE_WAY_HANDLE_DIRECTION_STATE = "core:ThreeWayHandleDirectionState"
-CORE_VIBRATION_STATE = "core:VibrationState"
-CORE_WATER_DETECTION_STATE = "core:WaterDetectionState"
-
-DEVICE_CLASS_BUTTON = "button"
-DEVICE_CLASS_GAS = "gas"
-DEVICE_CLASS_RAIN = "rain"
-DEVICE_CLASS_WATER = "water"
-
-ICON_WATER = "mdi:water"
-ICON_WATER_OFF = "mdi:water-off"
-ICON_WAVES = "mdi:waves"
-ICON_WEATHER_RAINY = "mdi:weather-rainy"
-
-IO_VIBRATION_DETECTED_STATE = "io:VibrationDetectedState"
-
 STATE_OPEN = "open"
 STATE_PERSON_INSIDE = "personInside"
 STATE_DETECTED = "detected"
-STATE_PRESSED = "pressed"
-
-TAHOMA_BINARY_SENSOR_DEVICE_CLASSES = {
-    "AirFlowSensor": DEVICE_CLASS_GAS,
-    "CarButtonSensor": DEVICE_CLASS_BUTTON,
-    "ContactSensor": DEVICE_CLASS_OPENING,
-    "MotionSensor": DEVICE_CLASS_MOTION,
-    "OccupancySensor": DEVICE_CLASS_OCCUPANCY,
-    "RainSensor": DEVICE_CLASS_RAIN,
-    "SirenStatus": DEVICE_CLASS_OPENING,
-    "SmokeSensor": DEVICE_CLASS_SMOKE,
-    "WaterDetectionSensor": DEVICE_CLASS_WATER,
-    "WaterSensor": DEVICE_CLASS_WATER,
-    "WindowHandle": DEVICE_CLASS_OPENING,
-}
+
+
+@dataclass
+class OverkizBinarySensorDescription(BinarySensorEntityDescription):
+    """Class to describe a Overkiz binary sensor."""
+
+    is_on: Callable[[Any], Any] = lambda state: state
+
+
+BINARY_SENSOR_DESCRIPTIONS = [
+    # RainSensor/RainSensor
+    OverkizBinarySensorDescription(
+        key="core:RainState",
+        name="Rain",
+        icon="mdi:weather-rainy",
+        is_on=lambda state: state == STATE_DETECTED,
+    ),
+    # SmokeSensor/SmokeSensor
+    OverkizBinarySensorDescription(
+        key="core:SmokeState",
+        name="Smoke",
+        device_class=binary_sensor.DEVICE_CLASS_SMOKE,
+        is_on=lambda state: state == STATE_DETECTED,
+    ),
+    # WaterSensor/WaterDetectionSensor
+    OverkizBinarySensorDescription(
+        key="core:WaterDetectionState",
+        name="Water",
+        icon="mdi:water",
+        is_on=lambda state: state == STATE_DETECTED,
+    ),
+    # AirSensor/AirFlowSensor
+    OverkizBinarySensorDescription(
+        key="core:GasDetectionState",
+        name="Gas",
+        device_class=binary_sensor.DEVICE_CLASS_GAS,
+        is_on=lambda state: state == STATE_DETECTED,
+    ),
+    # WindowHandle/ThreeWayWindowHandle
+    OverkizBinarySensorDescription(
+        key="core:OpenClosedState",
```

Comment:
This one triggers for my cover as well.. Should we keep it and add a binary sensor next to the cover implementation? 

<img width="332" alt="Screen Shot 2021-08-09 at 17 05 07" src="https://user-images.githubusercontent.com/1424596/128728850-fc77415a-8a3a-4b36-b30f-d7af5387da12.png">

Reply:
@tetienne

Reply:
I would maybe be in favor of removing `core:OpenClosedTiltState`, `core:OpenClosedState` and `core:OpeningState` since this should be part of cover already. My pergola also has `core:SlatsOpenClosedState` and we don't include this as well.

Reply:
That's OK for me. It will avoid to spam the device with too much entity.

Reply:
Great! And if people miss it and report it, we can always see what to do :-).